src/ir_passes/await.ml

```diff
@@ -416,19 +446,24 @@ and c_exp' context exp k =
   | PrimE (AwaitPrim s, [exp1]) ->
     let r = match LabelEnv.find_opt Throw context with
       | Some (Cont r) -> r
-      | Some Label
-      | None -> assert false
+      | _ -> assert false
     in
+    let b = match LabelEnv.find_opt Cleanup context with
+      | Some (Cont r) -> r
+      | None -> ContVar (var "@cleanup" bail_contT)
```

Comment:
Actually, can this case (`None`) ever arise? Aren't we always in the context of an AsyncE/AsyncE*, in which case the env  shouldn't contain a Cleanup? Maybe this is a hangover from the code you had before you extended AsyncE*?